### PR TITLE
net-im/synapse: add 1.66.0

### PR DESCRIPTION
**Pull request by arkamar**
There are two differences from previous releases:
- new dependency on `dev-python/pydantic`
- working tests with postgresql database, when `postgres` use flag is enabled

### CODE REVIEW

File: net-im/synapse/synapse-1.66.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+			dev-python/parameterized[${PYTHON_USEDEP}]
+		')
+		postgres? ( dev-db/postgresql[server] )
+	)
+"
+
+src_test() {
+	if use postgres; then
+		initdb -D "${T}/pgsql" || die
+		pg_ctl -w -D "${T}/pgsql" start \
```
Comment (thesamesam) [inline net-im/synapse/synapse-1.66.0.ebuild:85]: If `pg_ctl` supports long command options, we should perhaps use them here to ease readability.

File: net-im/synapse/synapse-1.66.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	fowners synapse:synapse /var/{lib,log}/synapse /etc/synapse
+	fperms 0750 /var/{lib,log}/synapse /etc/synapse
+	newinitd "${FILESDIR}/${PN}.initd" "${PN}"
+	systemd_dounit "${FILESDIR}/synapse.service"
+}
+
+pkg_postinst() {
+	optfeature "VoIP relaying on your homeserver with turn" net-im/coturn
+
+	einfo "In order to generate initial configuration run:"
```
Comment (thesamesam) [inline net-im/synapse/synapse-1.66.0.ebuild:116]: Something to consider: messages like this are sometimes only shown on first install instead. See https://devmanual.gentoo.org/ebuild-writing/variables/index.html's comment on REPLACING_VERSIONS (but there's actual examples in tree).

But that's not me saying you must change it now.

File: net-im/synapse/synapse-1.66.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+src_test() {
+	if use postgres; then
+		initdb -D "${T}/pgsql" || die
+		pg_ctl -w -D "${T}/pgsql" start \
+			-o "-h '' -k '${T}'" || die
+		createdb -h "${T}" synapse_test || die
+
+		local -x SYNAPSE_POSTGRES=1
+		local -x SYNAPSE_POSTGRES_HOST="${T}"
```
Comment (thesamesam) [inline net-im/synapse/synapse-1.66.0.ebuild:90]: Oh, and this is a mild suggestion: I try to link to e.g. CI configs / docs on test configuration if any exists upstream. If there isn't any, so be it, but if there's some place where the various options to control tests are listed, pop a link to them (e.g. where one might find SYNAPSE_PROGRES_HOST). However, by the sounds of it, it's just a generic option which happens to then influence tests (rather htan it being in some specific test driver section or whatever).

File: net-im/synapse/synapse-1.66.0.ebuild (hunk 1 above)
Comment (arkamar) [reply net-im/synapse/synapse-1.66.0.ebuild:85]: Yes, it supports long versions as well. I actually used as an inspiration couple of other ebuilds and copy-pasted this part. I'll change it with next release.

File: net-im/synapse/synapse-1.66.0.ebuild (hunk 2 above)
Comment (arkamar) [reply net-im/synapse/synapse-1.66.0.ebuild:116]: Thanks, I'll use it next time.

File: net-im/synapse/synapse-1.66.0.ebuild (hunk 3 above)
Comment (arkamar) [reply net-im/synapse/synapse-1.66.0.ebuild:90]: Ok, will do. Thanks.

**Pull request closed without merging**